packages/dds/tree/src/feature-libraries/object-forest/objectForest.ts

```diff
@@ -398,18 +398,25 @@ class Cursor implements ITreeSubscriptionCursor {
         return { result: TreeNavigationResult.NotFound, moved: 0 };
     }
     up(): TreeNavigationResult {
-        if (this.parentStack.pop() === undefined) {
-            // We are at the root, so return NotFound without making any changes to the state.
+        const length = this.parentStack.length;
+        assert(this.indexStack.length === length, "Unexpected indexStack.length");
+        assert(this.keyStack.length === length - 1, "Unexpected keyStack.length");
+
+        // Already at the root
+        if (length === 1) {
             return TreeNavigationResult.NotFound;
         }
+
+        assert(length > 1, "Unexpected parentStack.length");
+        this.parentStack.pop();
         this.indexStack.pop();
         this.keyStack.pop();
         // TODO: maybe compute siblings lazily or store in stack? Store instead of keyStack?
-        if (this.parentStack.length === 0) {
+        if (length === 2) {
```

Comment:
nit: This also required a double take for me. I think either adding a comment or assigning meaning to this value would help.

Reply:
comments added.